File: app/admin/router/system/official.go
package system

import (
	"github.com/gin-gonic/gin"
	"haocean/health-enforcement/app/admin/api/system"
	"haocean/health-enforcement/app/core/utils/jwt"
)

// InitOfficial 初始化执法人员路由
func InitOfficial(e *gin.Engine) {
	v := e.Group("system")
	{
		v.GET("/official/list", system.ListOfficial)
		v.GET("/official/:id", system.GetOfficial)

		auth := v.Group("")
		auth.Use(jwt.JWTAuthMiddleware())
		{
			auth.POST("/official", system.AddOfficial)
			auth.PUT("/official", system.UpdateOfficial)
			auth.DELETE("/official/:ids", system.DeleteOfficial)
			auth.POST("/official/bind-device", system.BindDevice)
		}
	}
}

File: frontend/health-enforcement-ui/src/api/system/official.js
import request from '@/utils/request'
import { parseStrEmpty } from "@/utils/ruoyi";

// 查询执法人员列表
export function listOfficial(query) {
  return request({
    url: '/system/official/list',
    method: 'get',
    params: query
  })
}

// 查询执法人员详细
export function getOfficial(officialId) {
  return request({
    url: '/system/official/' + parseStrEmpty(officialId),
    method: 'get'
  })
}

// 新增执法人员
export function addOfficial(data) {
  return request({
    url: '/system/official',
    method: 'post',
    data: data
  })
}

// 修改执法人员
export function updateOfficial(data) {
  return request({
    url: '/system/official',
    method: 'put',
    data: data
  })
}

// 删除执法人员
export function deleteOfficial(officialIds) {
  return request({
    url: '/system/official/' + officialIds,
    method: 'delete'
  })
}

// 绑定设备
export function bindDevice(data) {
  return request({
    url: '/system/official/bind-device',
    method: 'post',
    data: data
  })
}
